### PR TITLE
Ported unpack from well known types to resolve protobuf dependency error.

### PR DESCRIPTION
**Pull request by landrito**

### CODE REVIEW

File: spec/google/gax/operation_spec.rb [hunk 1]
```diff
@@ -33,7 +33,6 @@
 require 'google/gax/settings'
 require 'google/gax/constants'
 require 'google/protobuf/any_pb'
```
Comment (swcloud) [inline spec/google/gax/operation_spec.rb]: Does this line still need to depend on google-protobuf?

Comment (landrito) [inline spec/google/gax/operation_spec.rb]: Ah you're right. I just made gax-ruby depend on 3.0.2. Is this okay?

Comment (swcloud) [inline spec/google/gax/operation_spec.rb]: That should be fine. But let us hold off this PR until we know the release date of grpc 1.1.

Comment (landrito) [inline spec/google/gax/operation_spec.rb]: SGTM

Comment (landrito) [inline spec/google/gax/operation_spec.rb]: Any updates on this @swcloud?

Comment (swcloud) [inline spec/google/gax/operation_spec.rb]: Not yet.

Comment (swcloud) [inline spec/google/gax/operation_spec.rb]: @landrito Please go ahead with this change. We cannot wait for the new grpc gem any more.

File: spec/google/gax/operation_spec.rb [hunk 2]
```diff
@@ -57,13 +56,26 @@ def cancel_operation(name)
...
+# remove this function and use Google::Protobuf::Any#pack.
+def pack(any_pb, msg, type_url_prefix = 'type.googleapis.com/')
+  if type_url_prefix.empty? || type_url_prefix[-1] != '/'
+    any_pb.type_url = "#{type_url_prefix}/#{msg.class.descriptor.name}"
+  else
+    any_pb.type_url = "#{type_url_prefix}#{msg.class.descriptor.name}"
+  end
+  any_pb.value = msg.to_proto
+end
+
```
Comment (swcloud) [inline spec/google/gax/operation_spec.rb]: Add 'private :pack'?

Comment (landrito) [inline spec/google/gax/operation_spec.rb]: This is a test file so that shouldn't be necessary.

**Pull request merged**